projects/js-packages/publicize-components/src/components/form/connections-list.tsx

```diff
@@ -7,12 +9,22 @@ import styles from './styles.module.scss';
 import { useConnectionState } from './use-connection-state';
 
 export const ConnectionsList: React.FC = () => {
+	const { recordEvent } = useAnalytics();
+
 	const { connections, toggleById } = useSocialMediaConnections();
 
 	const { canBeTurnedOn, shouldBeDisabled } = useConnectionState();
 
 	const { needsUserConnection } = usePublicizeConfig();
 
+	const toggleConnection = useCallback(
+		( connectionId: string ) => () => {
+			toggleById( connectionId );
+			recordEvent( 'jetpack_social_connection_toggled', { location: 'editor' } );
```

Comment:
Apart from `location`, should we also add the connection data like `service_name`, whether enabled or disabled etc. to the data to see which connections are being toggled.